src/geocat/comp/interpolation.py

```diff
@@ -134,17 +134,93 @@ def _vertical_remap(func_interpolate, new_levels, xcoords, data, interp_axis=0):
+    R_d = 287.04  # dry air gas constant
+    g_inv = 1 / 9.80616  # inverse of gravity
+    alpha = 0.0065 * R_d * g_inv
+
+    tstar = t_bot * (1 + alpha * (ps / p_sfc - 1))
+    hgt = phi_sfc * g_inv
+    t0 = tstar + 0.0065 * hgt
+
+    alph = xr.where((tstar <= 290.5) & (t0 > 290.5),
+                    R_d / phi_sfc * (290.5 - tstar), alpha)
+
+    alph = xr.where((tstar > 290.5) & (t0 > 290.5), 0, alph)
+    tstar = xr.where((tstar > 290.5) & (t0 > 290.5), 0.5 * (290.5 + tstar),
+                     tstar)
+
+    tstar = xr.where((tstar < 255), 0.5 * (tstar + 255), tstar)
+
+    alnp = alph * np.log(lev / ps)
+    return hgt - R_d * tstar * g_inv * np.log(
+        lev / ps) * (1 + 0.5 * alnp + 1 / 6 * alnp**2)
+
+
+def _vertical_remap_extrap(new_levels, lev_dim, data, output, pressure, ps,
+                           variable, t_bot, phi_sfc):
+    # TODO: check for appropriate input values
+    plev_name = pressure.cf['vertical'].name
+    sfc_index = pressure[plev_name].argmax().data  # index of the model surface
+    p_sfc = pressure.isel(**dict({plev_name: sfc_index
+                                 }))  # extract pressure at lowest level
+
+    if variable == 'temperature':
+        for lev in new_levels:
+            output.loc[dict(plev=lev)] = xr.where(
+                lev <= p_sfc, output.sel(plev=lev),
+                _temp_extrapolate(data, lev_dim, lev, p_sfc, ps, phi_sfc))
+
+    elif variable == 'geopotential':
+        for lev in new_levels:
+            output.loc[dict(plev=lev)] = xr.where(
+                lev <= p_sfc, output.sel(plev=lev),
+                _geo_height_extrapolate(t_bot, lev, p_sfc, ps, phi_sfc))
+    else:
+        for lev in new_levels:
+            output.loc[dict(plev=lev)] = xr.where(
+                lev <= p_sfc, output.sel(plev=lev),
+                data.isel(**dict({plev_name: sfc_index})))
+    return output
+
+
 def interp_hybrid_to_pressure(data: xr.DataArray,
                               ps: xr.DataArray,
                               hyam: xr.DataArray,
                               hybm: xr.DataArray,
                               p0: float = 100000.,
                               new_levels: np.ndarray = __pres_lev_mandatory__,
                               lev_dim: str = None,
-                              method: str = 'linear') -> xr.DataArray:
-    """Interpolate data from hybrid-sigma levels to isobaric levels. Keeps
-    attributes (i.e. meta information) of the input data in the output as
-    default.
+                              method: str = 'linear',
+                              extrapolate: bool = False,
+                              variable: str = None,
+                              t_bot: xr.DataArray = None,
+                              phi_sfc: xr.DataArray = None) -> xr.DataArray:
+    """Interpolate and extrapolate data from hybrid-sigma levels to isobaric
+    levels. Keeps attributes (i.e. meta information) of the input data in the
```

Comment:
This is picky, feel free to disagree
```suggestion
    levels. Keeps attributes (i.e. metadata) of the input data in the
```